libavcodec/vaapi_encode_h264.c

```diff
@@ -833,7 +835,7 @@ static void vaapi_encode_h264_default_ref_pic_list(AVCodecContext *avctx,
                 break;
         }
         if (i == n)
-            FFSWAP(VAAPIEncodePicture*, rpl1[0], rpl1[1]);
+            FFSWAP(HWBaseEncodePicture*, rpl1[0], rpl1[1]);
```

Comment:
Preper to keep VAAPIEncodePicture as parameter type of this api which will benefit to keep it origin logic. Like here, only swap base pic is incorrect. all pic contents should be swaped.

Reply:
Base pic and vaapi share the same address. So I guess maybe it's ok? Anyway I changed it to the original logic and added  pointer cast after it.
Like this `if (base_pic->refs[0][i] != (HWBaseEncodePicture *)def_l0[i])`